src/postal/resource.py

```diff
@@ -238,7 +238,7 @@ def _model(data, fields=None):
                     url_id, fields = handler.resource_uri(data)
 
                     try:
-                        ret['resource_uri'] = permalink(lambda: (url_id, fields))()
+                        ret['resource_uri'] = reverse(lambda: (url_id, fields))()
```

Comment:
This isn't the correct way to replace permalink with reverse. permalink is a decorator, which means it takes a function as its argument and then returns another function. Decorators are not specific to Django, they're a python concept. Here's a decent intro to python decorators: https://www.thecodeship.com/patterns/guide-to-python-function-decorators/

lambda is a python keyword that creates a function that returns the value to the right of the colon using any parameters defined to the left of the colon (although there are not parameters here).  Learn more here: https://www.w3schools.com/python/python_lambda.asp

The lambda is acting like the url function in the example in the deprecation notice here:
https://docs.djangoproject.com/en/2.2/releases/1.11/#models-permalink-decorator

You'll notice there's an extra set of parentheses after the permalink  - that's where the function returned by lambda and then returned by permalink is then being called. reverse doesn't return a function, so we definitely don't want to have those at the end.

So this line should instead be 
ret['resource_uri'] = reverse(url_id, fields)

See if you can find a way to run a test suite for this to ensure that this code is exercised somewhere both before and after your fix.

Reply:
FYI - I managed to get the test sujite working but it doesn't actually test that code, so don't worry about testing it here much, just spend some time to understand the fix and then make it.

Reply:
Okay sounds good! I was trying to get the test suite working for so long and then I just ended up going back to my django 2.2 work. I feel really dumb for making this mistake. I just read online that we could swap permalink with reverse and thats what I ended up doing. I should look more carefully at what the code actually does.